test/__init__.py

```diff
@@ -1051,3 +1111,95 @@ def print_running_clients():
                 processed.add(obj._topology_id)
         except ReferenceError:
             pass
+
+
+def _all_users(db):
+    return {u["user"] for u in (db.command("usersInfo")).get("users", [])}
+
+
+def _create_user(authdb, user, pwd=None, roles=None, **kwargs):
+    cmd = SON([("createUser", user)])
+    # X509 doesn't use a password
+    if pwd:
+        cmd["pwd"] = pwd
+    cmd["roles"] = roles or ["root"]
+    cmd.update(**kwargs)
+    return authdb.command(cmd)
+
+
+def connected(client):
+    """Convenience to wait for a newly-constructed client to connect."""
+    with warnings.catch_warnings():
+        # Ignore warning that ping is always routed to primary even
+        # if client's read preference isn't PRIMARY.
+        warnings.simplefilter("ignore", UserWarning)
```

Comment:
I've never seen this warning. What is it referring to?

Reply:
This already exists in our test suite: https://github.com/mongodb/mongo-python-driver/blob/554ce7d9844ea3b49f71a7abb6c5ad6101dea131/test/utils.py#L866

Reply:
I did some digging and that warning was removed in https://jira.mongodb.org/browse/PYTHON-814 in PyMongo 3.0. Could you remove the catch_warnings block?